src/API/Site/Controllers/MerchantCenter/PriceBenchmarksController.php

```diff
@@ -84,19 +88,53 @@ protected function get_price_benchmarks_callback(): callable {
 				/** @var MerchantPriceBenchmarks $merchant */
 				$merchant = $this->container->get( MerchantPriceBenchmarks::class );
 
+				$merchant_report_data = $this->get_products_report_data( $request );
+
 				$benchmark_data      = $merchant->get_benchmark_data( $this->prepare_query_arguments( $request ) );
 				$price_insights_data = $merchant->get_price_insights( $this->prepare_query_arguments( $request ) );
 
 				// Map the data to the required format.
-				$response_data = $this->map_price_benchmarks_response( $benchmark_data, $price_insights_data );
-
+				$response_data = $this->map_price_benchmarks_response( $benchmark_data, $price_insights_data, $merchant_report_data );
 				return new Response( $response_data );
 			} catch ( Exception $e ) {
 				return $this->response_from_exception( $e );
 			}
 		};
 	}
 
+	/**
+	 * Retrieves the products report data for the given request.
+	 *
+	 * @param Request $request REST Request.
+	 * @return array|WP_Error Prepared response data or error.
+	 */
+	protected function get_products_report_data( Request $request ): array {
+		try {
+			/** @var MerchantReport $merchant_report */
+			$merchant_report         = $this->container->get( MerchantReport::class );
+			$prepare_query_arguments = $this->prepare_query_arguments( $request );
+			$product_id              = $request->get_param( 'id' );
+
+			// If a product ID is provided, set it in the query arguments.
+			if ( $product_id ) {
```

Comment:
Should probably bail early if there is no `$project_id` rather than still performing a query.